### PR TITLE
Add kotlinx.cinterop.getOriginalKotlinClass(objCClass/objCProtocol)

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**
Enables getting KClass for Kotlin classes produced to Objective-C header.

Also add other minor improvements.

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ObjCInterop.kt [hunk 1]
```diff
@@ -61,6 +63,9 @@ fun IrClass.isExternalObjCClass(): Boolean = this.isObjCClass() &&
             it.descriptor.annotations.hasAnnotation(externalObjCClassFqName)
         }
 
+fun ClassDescriptor.isObjCForwardDeclaration(): Boolean =
+        this.findPackage().fqName.startsWith(objcnamesForwardDeclarationsPackageName)
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ObjCInterop.kt]: Maybe check that it starts with `objcnames.`.

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ObjCInterop.kt]: This is exactly what is done here.

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ObjCInterop.kt]: `FqName.startsWith()` takes dots into account properly.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/ObjCStubs.kt [hunk 2]
```diff
@@ -690,3 +695,8 @@ private fun genProtocolGetter(
 
     return functionName
 }
+
+private fun ObjCClassOrProtocol.isProtocolClass(): Boolean = when (this) {
+    is ObjCClass -> (name == "Protocol" || binaryName == "Protocol")
+    is ObjCProtocol -> false
```
Comment (olonho) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/ObjCStubs.kt]: Interestingly, so what does it actually tests for?

Comment (SvyatoslavScherbina) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/ObjCStubs.kt]: That this is `Protocol` class (not the same as protocol).
`@protocol(Foo)` has `Protocol*` type.

**Pull request merged**